packages/stargate/src/stargateclient.ts

```diff
@@ -268,7 +270,34 @@ export class StargateClient {
     if (this.tmClient) this.tmClient.disconnect();
   }
 
-  public async broadcastTx(tx: Uint8Array): Promise<BroadcastTxResponse> {
+  public async broadcastTx(
+    tx: Uint8Array,
+    timeoutMs = 60_000,
+    pollIntervalMs = 3_000,
+  ): Promise<BroadcastTxResponse> {
+    return new Promise((resolve, reject) => {
+      const rejectTimeout = setTimeout(reject, timeoutMs, `timeout of ${timeoutMs}ms exceeded`);
+      this.broadcastTxCommit(tx)
+        .catch((error) => {
+          try {
+            const errorJson = JSON.parse(error.message);
+            if (errorJson.code !== -32603) {
+              throw error;
+            }
+            // timed out waiting for tx to be included in a block
+            // now we poll to artificially extend the timeout
+            const txHash = toHex(sha256(tx));
+            return this.pollTxById(txHash, pollIntervalMs);
```

Comment:
Will this recursive poll stack ever end after the timeout? If I understand correctly this continues in the background.

Reply:
Good point!